src/linker/Linker/CustomAttributeSource.cs

```diff
@@ -35,6 +66,8 @@ public IEnumerable<CustomAttribute> GetCustomAttributes (ICustomAttributeProvide
 					yield return customAttribute;
 			}
 
+			EnsureProcessedAttributeXml (provider);
```

Comment:
Avoid an extra lookup in case the assembly has no attribute XML.

Reply:
I would like to optimize this in the follow-up when I address the special case for '*' in corelib attributes. It interacts with this change because both need some way to track whether corelib's XML has been applied to an assembly.